src/components/MarketPlace/MarketPlace.jsx

```diff
@@ -127,27 +132,55 @@ export default function MarketPlace(props) {
         return `${quantityWithUnits} ${unit}`
     }
 
+    const getSellerOrder = (curUsers) => {
+        let sellerNumProducts = curUsers.map((user, indx) => {
+            if (user.uid != currentUser.uid && user.data.sale) {
+                let sellerCount = 0
+                user.data.sale.forEach((item) => {
+                    // TODO: better nlp algo to match names
+                    if (userCart.find(prod => prod.name == item.name)) {
+                        sellerCount += 1
+                    }
+                })
```

Comment:
This could be simplified by using either `user.data.sale.filter(...).length` or `.reduce`